test/integration/service/loadbalancer_test.go

```diff
@@ -711,10 +742,25 @@ func Test_ServiceLoadBalancerIPMode(t *testing.T) {
 				t.Fatalf("unexpected load balancer status")
 			}
 
-			gotIngress := service.Status.LoadBalancer.Ingress[0]
-			if gotIngress.IP != tc.externalIP || !reflect.DeepEqual(gotIngress.IPMode, tc.expectedIPMode) {
-				t.Errorf("unexpected load balancer ingress, got ingress %v, expected IP %v, expected ipMode %v",
-					gotIngress, tc.externalIP, tc.expectedIPMode)
+			ingress := service.Status.LoadBalancer.Ingress[0]
+			if !apiequality.Semantic.DeepEqual(&ingress, &tc.expectedIngress) {
+				t.Errorf("expected Ingress %v, got IP %v",
+					ingress, tc.expectedIngress)
+				if ingress.IPMode != nil && tc.expectedIngress.IPMode != nil {
+					t.Logf("IPMode %v expected %v", *ingress.IPMode, *tc.expectedIngress.IPMode)
+				}
+			}
+
+			// mutate the service and check the status is preserved
+			newService := service.DeepCopy()
+			newService.Spec.Ports[0].Port = 443
+			service, err = client.CoreV1().Services(ns.Name).Update(ctx, newService, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Error updating test service: %v", err)
+			}
+
+			if len(service.Status.LoadBalancer.Ingress) == 0 {
```

Comment:
This seems lite on the verification. Is it ok to have more than 1 ingress? Do we expect the port to change to 443?

Reply:
good point, I was trying to reproduce an issue that wiped out the status completely, but let me add a better assertion

Reply:
updated